### PR TITLE
ICU-20627 Adding alias locales to res_index (ignore xml deprecates).

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20627
- [x] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/python/icutools/databuilder/__main__.py [hunk 1]
```diff
@@ -83,12 +83,6 @@
     default = False,
     action = "store_true"
 )
-flag_parser.add_argument(
-    "--ignore_xml_deprecates",
```
Comment (markusicu) [inline icu4c/source/python/icutools/databuilder/__main__.py]: Should we keep obsolote options for one or more releases as no-ops, with the help text giving the status?
It does not seem necessary to break users who specify the option.
Or better to make them fail fast so they know this is no longer an option?

Comment (sffc) [reply icu4c/source/python/icutools/databuilder/__main__.py]: I just added the option in #642.  It isn't in ICU 64.2.

Comment (sffc) [reply icu4c/source/python/icutools/databuilder/__main__.py]: I added the option in #642 to give users (Google) a way to disable the feature.  But now that I'm proposing to make this the default behavior, I'm deleting the option I added.  If we wanted to give a way for users to get the old behavior, we could make an option named `--use-xml-deprecates`.

**Comment by sffc** [comment]
I will look at the test failures on Tuesday.

File: icu4c/source/python/icutools/databuilder/__main__.py (hunk 1 above)
Comment (markusicu) [reply icu4c/source/python/icutools/databuilder/__main__.py]: ok no concern then. tnx.

**Pull request closed without merging**